From 558e7b4dcf66fda0e0144acc923e088ff20241f3 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Tue, 13 Dec 2016 16:41:12 +1000
Subject: [PATCH 0289/2094] drm/amd/display: assign correct enum for edp
 revision

There are 2 edp enum revisions, no idea why, drop one, and just
assign 1.1 to the default value.

Change-Id: Ibb45d55b5fcb2be34e3feadddb7ddae70f92cbe9
Signed-off-by: Dave Airlie <airlied@redhat.com>
Reviewed-by: Tony Cheng <Tony.Cheng@amd.com>
Acked-by: Harry Wentland <Harry.Wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/display/dc/core/dc_link_dp.c |  2 +-
 drivers/gpu/drm/amd/display/include/dpcd_defs.h  | 16 ----------------
 2 files changed, 1 insertion(+), 17 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/dc/core/dc_link_dp.c b/drivers/gpu/drm/amd/display/dc/core/dc_link_dp.c
index d3625c2..a4b6a6a 100644
--- a/drivers/gpu/drm/amd/display/dc/core/dc_link_dp.c
+++ b/drivers/gpu/drm/amd/display/dc/core/dc_link_dp.c
@@ -2129,7 +2129,7 @@ static void retrieve_link_cap(struct core_link *link)
 	link->dpcd_caps.panel_mode_edp =
 		edp_config_cap.bits.ALT_SCRAMBLER_RESET;
 
-	link->edp_revision = DPCD_EDP_REVISION_EDP_UNKNOWN;
+	link->edp_revision = EDP_REVISION_11;
 
 	link->public.test_pattern_enabled = false;
 	link->public.compliance_test_state.raw = 0;
diff --git a/drivers/gpu/drm/amd/display/include/dpcd_defs.h b/drivers/gpu/drm/amd/display/include/dpcd_defs.h
index 5fc5538..2dca4a4 100644
--- a/drivers/gpu/drm/amd/display/include/dpcd_defs.h
+++ b/drivers/gpu/drm/amd/display/include/dpcd_defs.h
@@ -144,20 +144,4 @@ enum dpcd_psr_sink_states {
 	PSR_SINK_STATE_SINK_INTERNAL_ERROR = 7,
 };
 
-/* This enum defines the Panel's eDP revision at DPCD 700h
- * 00h = eDP v1.1 or lower
- * 01h = eDP v1.2
- * 02h = eDP v1.3 (PSR support starts here)
- * 03h = eDP v1.4
- * If unknown revision, treat as eDP v1.1, meaning least functionality set.
- * This enum has values matched to eDP spec, thus values should not change.
- */
-enum dpcd_edp_revision {
-	DPCD_EDP_REVISION_EDP_V1_1 = 0,
-	DPCD_EDP_REVISION_EDP_V1_2 = 1,
-	DPCD_EDP_REVISION_EDP_V1_3 = 2,
-	DPCD_EDP_REVISION_EDP_V1_4 = 3,
-	DPCD_EDP_REVISION_EDP_UNKNOWN = DPCD_EDP_REVISION_EDP_V1_1,
-};
-
 #endif /* __DAL_DPCD_DEFS_H__ */
-- 
2.7.4

